### PR TITLE
Cool Dining UI changes to make UX radically more enjoyable

### PR DESCRIPTION
**Pull request by jonathanmelitski**
PR:
- Added a two-way scroll view for displaying sections.
- Revamped Individual dining section view
- Sorted dining stations based on what people actually want to see (beverages should be at the bottom compared to "grill")
- Sorted dining hours display to make more readable.

### CODE REVIEW

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift [hunk 1]
```diff
@@ -2,101 +2,105 @@
 //  DiningVenueDetailMenuView.swift
 //  PennMobile
 //
-//  Created by CHOI Jongmin on 23/6/2020.
```
Comment (anli5005) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift:5]: @01jongmin a challenger arises

File: Gemfile.lock [hunk 2]
```diff
@@ -276,6 +276,7 @@ GEM
 PLATFORMS
   universal-darwin-21
   universal-darwin-22
+  universal-darwin-23
```
Comment (anli5005) [inline Gemfile.lock:279]: Normally I'd ask to revert Gemfile.lock changes, but this seems fine

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift [hunk 3]
```diff
@@ -2,101 +2,105 @@
...
+    
+    /// Constraints of this function:
+    /// Need to know if a meal is currently going on, to return it.
+    /// If there is a meal today that is closest (utilities), return it.
+    /// If the selected date is not the current day, return the first menu.
+    /// If at any point, the list of menus is empty, return nil.
+    func getMenu() -> DiningMenu? {
+        if (menus.count == 0) { return nil }
+        
+        if (!Calendar.current.isDate(menuDate, inSameDayAs: Date())) {
```
Comment (anli5005) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift:51]: We should probably make this timezone-aware at some point (namely forcing the timezone to always be `America/New_York`) but we can deal with that in another PR (ideally we'd add an extension to `Calendar`)

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift [hunk 4]
```diff
@@ -2,101 +2,105 @@
...
                 }
-                .pickerStyle(SegmentedPickerStyle())
-                DiningMenuRow(diningMenu: menus[menuIndex])
-                    .transition(.opacity)
             }
+            
+//            .onChange(of: parentScrollOffset) { _ in
+//                print(proxy.size)
+//            }
+
```
Comment (anli5005) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift:88]:
```suggestion
```

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailView.swift [hunk 5]
```diff
@@ -10,6 +10,7 @@ import SwiftUI
 import Kingfisher
 import FirebaseAnalytics
 import PennMobileShared
+import WebKit
```
Comment (anli5005) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailView.swift:13]: Let's switch this to use the new `.safari` modifier in a future PR

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/MenuDisclosureGroup.swift [hunk 6]
```diff
@@ -176,21 +306,21 @@ extension AnyTransition {
...
+//            ScrollView {
+//                VStack {
+//                    DiningVenueDetailMenuView(menus: diningVenues.menus, id: 1)
+//                    Spacer()
+//                }
+//            }.navigationTitle("Dining")
+//            .padding()
+//        }
+//    }
+//}
```
Comment (anli5005) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/MenuDisclosureGroup.swift:323]:
```suggestion
```

File: PennMobileShared/Dining/DiningAPI.swift [hunk 7]
```diff
@@ -46,6 +49,7 @@ public class DiningAPI {
         guard let (data, _) = try? await URLSession.shared.data(from: URL(string: diningMenuUrl + dateStr + "/")!) else {
             return .failure(.serverError)
         }
+        print(String(data: data, encoding: .utf8)!)
```
Comment (anli5005) [inline PennMobileShared/Dining/DiningAPI.swift:52]:
```suggestion
```

File: PennMobileShared/Dining/DiningAPI.swift [hunk 8]
```diff
@@ -8,6 +8,7 @@
 
 import SwiftyJSON
 import Foundation
+import JavaScriptCore
```
Comment (anli5005) [inline PennMobileShared/Dining/DiningAPI.swift:11]:
```suggestion
```

File: PennMobileShared/Dining/Models/DiningVenue+Extensions.swift [hunk 9]
```diff
@@ -22,24 +22,34 @@ public extension VenueType {
...
+            day.date == dateFormatter.string(from: date)
+        })?.meals.sorted(by: { el1, el2 in
+            return el1.starttime > el2.starttime
+        }) ?? []
     }
 
     var isOpen: Bool {
-        guard let mealsToday = mealsToday else { return false }
-        for meal in mealsToday.meals where meal.isCurrentlyServing {
+        if (mealsToday == []) { return false }
```
Comment (anli5005) [inline PennMobileShared/Dining/Models/DiningVenue+Extensions.swift:42]:
```suggestion
        if mealsToday.isEmpty { return false }
```

And so on going forward

**Review by anli5005 — APPROVED** [review]
LGTM! Left some minor nits but won't block on any of those

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/MenuDisclosureGroup.swift [hunk 10]
```diff
@@ -9,80 +9,208 @@
...
+            Spacer()
         }
     }
 }
 
+struct DiningMenuSectionRow: View {
+    let station: DiningStation
+
+    var body: some View {
+        Text("a")
```
Comment (JHawk0224) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/MenuDisclosureGroup.swift:210]: Is this view still needed?

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/MenuDisclosureGroup.swift [hunk 11]
```diff
@@ -120,6 +248,8 @@ struct DiningStationItemRow: View {
                 withAnimation {
                     isExpanded.toggle()
                 }
+            }.onChange(of: isExpanded) { _ in
+                print(diningStationItem.desc)
```
Comment (JHawk0224) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/MenuDisclosureGroup.swift:252]: Do we want to keep this print?

File: PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift [hunk 12]
```diff
@@ -2,101 +2,100 @@
...
-                            .bold()
-                        Spacer()
-                        Image(systemName: "chevron.right")
-                    }
-                    .padding()
-                    .foregroundColor(.blue).font(Font.system(size: 24).weight(.bold))
+        LazyVStack(pinnedViews: [.sectionHeaders]) {
+            HStack {
+                if currentMenu != nil {
+                    Picker("Menu", selection: Binding($currentMenu)!) {
```
Comment (jonathanmelitski) [inline PennMobile/Dining/SwiftUI/Views/Venue/Detail View/DiningVenueDetailMenuView.swift:66]: #630 oh how we've grown

**Pull request merged**